src/CMakeLists.txt

```diff
@@ -17,12 +17,20 @@ if (GUI)
     if (WIN32)
         list (APPEND QBT_QT_COMPONENTS WinExtras)
     endif(WIN32)
+    if (APPLE)
+        list (APPEND QBT_GUI_OPTIONAL_LINK_LIBRARIES objc)
+        list (APPEND QBT_QT_COMPONENTS MacExtras)
+    endif (APPLE)
 endif (GUI)
 if (DBUS)
     list (APPEND QBT_QT_COMPONENTS DBus)
 endif (DBUS)
 find_package(Qt5 5.5.1 COMPONENTS ${QBT_QT_COMPONENTS} REQUIRED)
 
+if (GUI AND APPLE)
+    # Fix MOC inability to detect macOS
+    add_definitions(-DQ_OS_MAC)
```

Comment:
This is to workaround Qt bugs, examples: [one](https://bugreports.qt.io/browse/QTBUG-52687), [two](https://bugreports.qt.io/browse/QTBUG-58325). From what I can tell there are no proper solutions.

It is necessary for [this file](https://github.com/qbittorrent/qBittorrent/pull/6952/files#diff-cd08514626207f11a4741106092c8404R37) in particular, which is GUI only. Without this define MOC tries to generate systray functions, and even if I let it include QSystemTrayIcon it will later fail with linking errors.